### ISSUE TITLE
No grouping support when creating criteria from string expressions (hoa/ruler)

### ISSUE THREAD

**Issue by roomcays** [shadowhand/latitude#83]
It seems that `\Latitude\QueryBuilder\Ruler\Visitor()` does not support grouping at all.
I wrote grouping test for `\Latitude\QueryBuilder\CriteriaTest`, that fails:

```php
class CriteriaTest extends TestCase
{
    public function testGrouping()
    {
        $criteria = $this->factory->criteria('users.active = true and (users.id = 100 or users.role = "admin")');

        $this->assertSql('users.active = true and (users.id = ? or users.role = ?)', $criteria);
        $this->assertParams([100, 'admin'], $criteria);
    }
}
```

Test result:
```diff
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'users.active = true and (users.id = ? or users.role = ?)'
+'users.active = true and users.id = ? or users.role = ?'
```

The same query fetched directly from `\Hoa\Ruler\Ruler::interpret` gives:
```php
$model = new \Hoa\Ruler\Model();
$model->expression =
    $model->and(
        $model->{'='}(
            $model->variable('users')
                ->attribute('active'),
            true
        ),
        $model->or(
            $model->{'='}(
                $model->variable('users')
                    ->attribute('id'),
                100
            ),
            $model->{'='}(
                $model->variable('users')
                    ->attribute('role'),
                'admin'
            )
        )
    );
```

**Comment by shadowhand** [comment]
Strange. I don't personally have any reason to work on this, but if you find a solution I would love to see a PR!

**Comment by shadowhand** [comment]
Fixed in #84, from what I can tell.
